### PR TITLE
[NETBEANS-3929] Adding the ability to delete a tag from a remote repo.

### PR DESCRIPTION
**Pull request by dmochalov**
Adding the ability to delete a tag from a remote repository using integrated Git Client. Before a user could delete a branch but could not delete a tag. Now, when performing git push... a user can select locally deleted tag and Git Client will perform git push :refs/tags/{tag_name_to_delete}
I tested the implementation locally using GitHub. It looks like this:

image

It's my first PR so any feedback would be highly appreciated.

### CODE REVIEW

File: ide/git/src/org/netbeans/modules/git/ui/push/PushAction.java [hunk 1]
```diff
@@ -341,10 +344,20 @@ protected void logUpdates (File repository, Map<String, GitTransportUpdate> upda
...
-                            }));
+                            //tag deleting or updating
+                            if (update.getNewObjectId() == null && update.getOldObjectId() != null) {
+                                //deleting tag from remote
+                                logger.outputLine(NbBundle.getMessage(PushAction.class, "MSG_PushAction.updates.deleteTag", new Object[]{ //NOI18N
+                                    update.getRemoteName(),
+                                    update.getOldObjectId(),
+                                    update.getResult(),}));
+                            } else {
+                                //updating or adding tag to the remoteё
```
Comment (matthiasblaesing) [inline ide/git/src/org/netbeans/modules/git/ui/push/PushAction.java]: This should not be there: ё

File: ide/git/src/org/netbeans/modules/git/ui/push/PushBranchesStep.java [hunk 2]
```diff
@@ -98,6 +98,13 @@ public HelpCtx getHelp() {
         return new HelpCtx(PushBranchesStep.class);
     }
 
+    /**
+     * *
```
Comment (matthiasblaesing) [inline ide/git/src/org/netbeans/modules/git/ui/push/PushBranchesStep.java]: One star to much.

File: ide/git/src/org/netbeans/modules/git/ui/push/PushBranchesStep.java [hunk 3]
```diff
@@ -98,6 +98,13 @@ public HelpCtx getHelp() {
...
+     * @param cfg Congiruration of the remote repository including urls of
```
Comment (matthiasblaesing) [inline ide/git/src/org/netbeans/modules/git/ui/push/PushBranchesStep.java]: Please keep it on one line and fix typo: "Configuration of the remote repository including urls of remote"

File: ide/git/src/org/netbeans/modules/git/ui/push/PushBranchesStep.java [hunk 4]
```diff
@@ -98,6 +98,13 @@ public HelpCtx getHelp() {
...
+     * remote
+     * @param branches list of all brances in the remote repo
```
Comment (matthiasblaesing) [inline ide/git/src/org/netbeans/modules/git/ui/push/PushBranchesStep.java]: Fix typo: list of all branches in the remote repo

File: ide/git/src/org/netbeans/modules/git/ui/push/PushBranchesStep.java [hunk 5]
```diff
@@ -154,6 +162,8 @@ protected void perform () {
                             }
                         }
                         boolean preselected = !conflicted && updateNeeded;
+
+                        //add current brunch in the list for update or for adding
```
Comment (matthiasblaesing) [inline ide/git/src/org/netbeans/modules/git/ui/push/PushBranchesStep.java]: While brunch is nice, most probably branch was meant.

File: ide/git/src/org/netbeans/modules/git/ui/push/PushMapping.java [hunk 6]
```diff
@@ -224,18 +224,58 @@ boolean isCreateBranchMapping () {
...
     
     public static final class PushTagMapping extends PushMapping {
-        private final GitTag tag;
+        private final GitTag tag; //local tag
         private final boolean isUpdate;
-        
+        private final String remoteTagName;
+
+        /**
+         * *
```
Comment (matthiasblaesing) [inline ide/git/src/org/netbeans/modules/git/ui/push/PushMapping.java]: one star to many

File: ide/git/src/org/netbeans/modules/git/ui/push/PushMapping.java [hunk 7]
```diff
@@ -224,18 +224,58 @@ boolean isCreateBranchMapping () {
...
+            this.tag = null;
+            this.isUpdate = remoteName != null;
+            this.remoteTagName = remoteName;
+        }
+
+        /**
+         * Adding or updating tag in the remote repository
+         *
+         * @param tag representation of a local tag
+         * @param remoteName remote tag name, can be null. If null than we crate
```
Comment (matthiasblaesing) [inline ide/git/src/org/netbeans/modules/git/ui/push/PushMapping.java]: crate => create

File: ide/libs.git/src/org/netbeans/libs/git/GitTransportUpdate.java [hunk 8]
```diff
@@ -74,14 +74,29 @@
         this.type = getType(update.getLocalName());
     }
 
-    GitTransportUpdate (URIish uri, RemoteRefUpdate update, Map<String, GitBranch> remoteBranches) {
+    /**
+     * *
```
Comment (matthiasblaesing) [inline ide/libs.git/src/org/netbeans/libs/git/GitTransportUpdate.java]: to many stars.

**Comment by matthiasblaesing** [comment]
I assume, that you are aware of the license header in the edited files and thus agree that you are willing to donate the changeset to the ASF.

**Comment by dmochalov** [comment]
@matthiasblaesing , thank you for your response. I corrected the typos. Should i squash them to one commit?
Yes I am aware of the license header and agree to donate the changeset to the ASF.

**Comment by matthiasblaesing** [comment]
@dmochalov thank you - looks sane to me. Yes, please squash the commits, in theory github can do this when merging, but that requires the accounts from which the commits were created, to have public real name and email.

**Comment by matthiasblaesing** [comment]
Forget it - merged via 7d17072e22ca45228e57ef2399ecfd7ac37c53e2. Thank you

**Pull request closed without merging**